### PR TITLE
selftests/functional/test_output.py: fix file writing/reading race co…

### PR DESCRIPTION
**Pull request by clebergnu**
…ndition

The test_print_to_std test attempts to read from a file created by the
Avocado runner.  It was observed that in some situations it reads from
a partially flushed file (missing the last string).

Let's attempt to sync the file buffers into disk before reading from it.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

### CODE REVIEW

File: avocado/utils/path.py [hunk 1]
```diff
@@ -191,3 +192,21 @@ def usable_ro_dir(directory):
...
+def sync_attempt():
+    """
+    Attempts (best-effort) to synchronize file buffers into disk
+
+    This method attempts to to call the native Python standard library
+    method where available, and then falls back to a ctypes wrapper when
+    necessary.
+    """
+    if hasattr(os, 'sync'):
+        os.sync()
```
Comment (apahim) [inline avocado/utils/path.py]: We already rely on `os.sync()` in `jobdata` module. If there's a problem with it's availability, we have to address that creating the fallback everywhere (i.e. https://github.com/avocado-framework/avocado/blob/master/avocado/core/jobdata.py#L86)

Comment (clebergnu) [reply avocado/utils/path.py]: `fsync()` is available on all Python versions, it's `sync()` that's not.

Comment (apahim) [reply avocado/utils/path.py]: ops :)
then it looks good.

**Comment by clebergnu** [comment]
This may mitigate the issue, but it doesn't solve the root cause.  I have come up with a reproducer now, which needs a different fix.

**Pull request closed without merging**